lib/runtime/src/component.rs

```diff
@@ -48,6 +48,7 @@ use super::{
 };
 
 use crate::pipeline::network::{ingress::push_endpoint::PushEndpoint, PushWorkHandler};
+use crate::protocols::Endpoint as EndpointId;
```

Comment:
Is `protocols::Endpoint` something you think we should rename for clarity / less conflicts with other "Endpoint" definitions?

Reply:
Absolutely. I started renaming it to `EndpointId` everywhere but backed it out because it was too big for this PR.

Reply:
Sounds good to me!